test/RelayPricer.test.ts

```diff
@@ -1,180 +1,204 @@
-//             stubPricer = stub(pricer, 'findAvailableApi')
-//                 .withArgs(RBTC_SYMBOL)
-//                 .returns(fakeCoinbaseApi)
-//                 .withArgs(RDOC_SYMBOL)
-//                 .returns(fakeRdocApi);
-
-//             const price = await pricer.getExchangeRate(
-//                 RDOC_SYMBOL,
-//                 RBTC_SYMBOL
-//             );
-
-//             expect(price.toString()).to.equal(X_RATE_RDOC_RBTC.toString());
-//         });
-
-//         it('should return target exchange rate', async () => {
-//             buildExchangeRate(X_RATE_RBTC_USD, X_RATE_RIF_USD);
-//             const price = await pricer.getExchangeRate(RBTC_SYMBOL, RIF_SYMBOL);
-//             assert.equal(
-//                 X_RATE_RIF_RBTC.toString(),
-//                 price.toString(),
-//                 'price is not the same'
-//             );
-//         });
-
-//         it('should fail if source currency is not valid', async () => {
-//             const error = new Error(
-//                 `Coinbase API does not recognise given currency ${RIF_SYMBOL}`
-//             );
-//             fakeSourceApi.queryExchangeRate.throws(error);
-//             await assert.isRejected(
-//                 pricer.getExchangeRate(RIF_SYMBOL, 'NA', USD_SYMBOL),
-//                 error.message
-//             );
-//         });
-
-//         it('should fail if intermediate currency is not valid', async () => {
-//             const error = new Error(
-//                 `Exchange rate for currency pair ${RIF_SYMBOL}/${RBTC_SYMBOL} is not available`
-//             );
-//             fakeSourceApi.queryExchangeRate.throws(error);
-//             await assert.isRejected(
-//                 pricer.getExchangeRate(RIF_SYMBOL, RBTC_SYMBOL, 'NA'),
-//                 error.message
-//             );
-//         });
-
-//         it('should fail if target exchange rate is 0', async () => {
-//             buildExchangeRate(X_RATE_RBTC_USD, new BigNumber(0));
-//             await assert.isRejected(
-//                 pricer.getExchangeRate(RIF_SYMBOL, RBTC_SYMBOL),
-//                 conversionError
-//             );
-//         });
-
-//         it('should fail if source exchange rate is 0', async () => {
-//             buildExchangeRate(new BigNumber(0), X_RATE_RIF_USD);
-//             await assert.isRejected(
-//                 pricer.getExchangeRate(RIF_SYMBOL, RBTC_SYMBOL),
-//                 conversionError
-//             );
-//         });
-//     });
-// });
+import chai, { assert, expect } from 'chai';
+import { SinonStub, SinonStubbedInstance, stub, restore } from 'sinon';
+import chaiAsPromised from 'chai-as-promised';
+import { BigNumber as BigNumberJs } from 'bignumber.js';
+
+import RelayPricer from '../src/RelayPricer';
+import type BaseExchangeApi from '../src/api/pricer/ExchangeApi';
+
+chai.use(chaiAsPromised);
+
+const randomBigNumberJs = (base: number) =>
+    BigNumberJs((Math.random() * base));
```

Comment:
NIT: I'd change the argument name to be `max` and furthermore, I don't think we need the double parentheses.
```suggestion
const randomBigNumberJs = (max: number) =>
    BigNumberJs(Math.random() * max);
```

Reply:
updated